libsupport/include/katana/Result.h

```diff
@@ -260,6 +260,40 @@ ResultErrno() {
   return std::error_code(errno, std::system_category());
 }
 
+template <class>
+struct is_result : std::false_type {};
+
+template <class T>
+struct is_result<Result<T>> : std::true_type {};
+
+template <class T>
+std::enable_if_t<!std::is_same<T, void>::value, T&&>
+extract_value(Result<T>&& result) {
+  return std::move(result.value());
+}
+
+inline int
+extract_value(Result<void>&&) {
+  return 0;
+}
+
+#define KATANA_CHECK_NAME(x, y) x##y
+
+#define KATANA_CHECK_IMPL(result_name, expression)                             \
+  ({                                                                           \
+    auto result_name = (expression);                                           \
+    static_assert(                                                             \
+        ::katana::is_result<decltype(result_name)>::value,                     \
+        "KATANA_CHECK requires a katana::Result");                             \
```

Comment:
Can we avoid the need for this by have a function (maybe just an identity function) that we pass the value to. That would trigger a more normal and complete type error.

Reply:
This was, in some sense, already resolved by the function extracting the value from the Result, so I've removed the static_assert